bin/nvcc_wrapper

```diff
@@ -224,7 +224,7 @@ do
     std_flag=$corrected_std_flag
     shared_args="$shared_args $std_flag"
   ;;
-  --std=c++11|-std=c++11|--std=c++14|-std=c++14)
+  --std=c++11|-std=c++11|--std=c++14|-std=c++14|--std=c++17|-std=c++17)
```

Comment:
If you build with, e.g. CUDA 10 and GCC 8, CMake expects to have C++17 support but nvcc won't allow it. This undoes our trick of converting c++17 to c++14 for CMake feature tests.

Reply:
This is fixed in CMake 3.17 after they changed how feature detection works. This doesn't work in < 3.16, which is what we are stuck on.

Reply:
I'm guessing we don't have a CMake < 3.16, GCC 8, CUDA 10 build in Jenkins/Travis

Reply:
Hmm... I still can't reproduce any problem:
```
$ cmake --version
cmake version 3.10.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```
```
$ cmake -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON -DKokkos_CXX_STANDARD=14 ..
-- The CXX compiler identification is GNU 8.4.0
-- Check for working CXX compiler: /tmp/kokkos_new/bin/nvcc_wrapper
-- Check for working CXX compiler: /tmp/kokkos_new/bin/nvcc_wrapper -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The project name is: Kokkos
-- Compiler Version: 10.2.89
-- SERIAL backend is being turned on to ensure there is at least one Host space. To change this, you must enable another host execution space and configure with -DKokkos_ENABLE_SERIAL=OFF or change CMakeCache.txt
-- Using -std=c++14 for C++14 standard as feature
-- Execution Spaces:
--     Device Parallel: CUDA
--     Host Parallel: NONE
--       Host Serial: SERIAL
-- 
-- Architectures:
--  VOLTA70
-- Found CUDAToolkit: /tmp/cuda-10.2/include (found version "10.2.89") 
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Found TPLCUDA: TRUE  
-- Found TPLLIBDL: /usr/lib/x86_64-linux-gnu/libdl.so  
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/kokkos_new/build_nvcc
```

Reply:
````
-- Compiler Version: 10.1.243
-- SERIAL backend is being turned on to ensure there is at least one Host space. To change this, you must enable another host execution space and configure with -DKokkos_ENABLE_SERIAL=OFF or change CMakeCache.txt
-- Using -std=c++11 for C++11 standard as feature
CMake Error at cmake/kokkos_test_cxx_std.cmake:47 (MESSAGE):
  Compiler NVIDIA should support cxx_std_11, but CMake reports feature not
  supported
Call Stack (most recent call first):
  cmake/kokkos_test_cxx_std.cmake:63 (kokkos_set_cxx_standard_feature)
  cmake/kokkos_tribits.cmake:201 (INCLUDE)
  CMakeLists.txt:161 (KOKKOS_SETUP_BUILD_ENVIRONMENT)

````
Fascinating. Does not happen with 3.10. Does happen with 3.12.

Reply:
Alright, I can confirm that it fails for 3.12, 3.13 and 3.14. I now pushed another case to `nvcc_wrapper` that handles `-std=c++17` for `NVCC` version < 11 and version >= 11.